pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -490,6 +492,58 @@ func (m *kubeGenericRuntimeManager) executePreStopHook(pod *v1.Pod, containerID
 	return int64(metav1.Now().Sub(start.Time).Seconds())
 }
 
+// executeRetryUntilSuccess runs the Retry-Until lifecycle hooks if applicable and returns the duration it takes.
+func (m *kubeGenericRuntimeManager) executeRetryUntilSuccessHook(pod *v1.Pod, containerID kubecontainer.ContainerID, containerSpec *v1.Container, gracePeriod int64) int64 {
+	glog.V(4).Infof("Running RetryUntilSuccess hook for container %q", containerID.String())
+
+	start := metav1.Now()
+	done := make(chan struct{})
+	controllerChan := make(chan struct{})
+	go wait.Until(func() {
```

Comment:
Actually, it looks like you don't need the done channel, so I think `wait.Poll` would be a better fit.